tiles/src/main/java/com/protomaps/basemap/layers/Pois.java

```diff
@@ -3,85 +3,308 @@
+
       if (sf.hasTag("aeroway", "aerodrome")) {
         kind = sf.getString("aeroway");
-      } else if (sf.hasTag("amenity", "cafe", "college", "hospital", "library", "post_office", "school", "townhall")) {
+        min_zoom = 13;
+
+        // Emphasize large international airports earlier
+        if( kind == "aerodrome" &&  sf.hasTag("iata")) {
+          min_zoom -= 2;
+        }
+
+        if( sf.hasTag("aerodrome") ) {
+          kind_detail = sf.getString("aerodrome");
+        }
+      } else if (sf.hasTag("amenity",  "university","college")) {
+        kind = sf.getString("amenity");
+        min_zoom = 11;
+      } else if (sf.hasTag("amenity", "hospital")) {
+        kind = sf.getString("amenity");
+        min_zoom = 12;
+      } else if (sf.hasTag("amenity", "library", "post_office", "school", "townhall")) {
         kind = sf.getString("amenity");
+        min_zoom = 13;
+      } else if (sf.hasTag("amenity", "cafe")) {
+        kind = sf.getString("amenity");
+        min_zoom = 15;
       } else if (sf.hasTag("landuse", "cemetery")) {
         kind = sf.getString("landuse");
+        min_zoom = 14;
       } else if (sf.hasTag("leisure", "golf_course", "marina", "park", "stadium")) {
         kind = sf.getString("leisure");
-      } else if (sf.hasTag("leisure")) {
-        // This is dubious but existing behavior
-        kind = "park";
+        min_zoom = 13;
       } else if (sf.hasTag("shop", "grocery", "supermarket")) {
         kind = sf.getString("shop");
+        min_zoom = 14;
       } else if (sf.hasTag("tourism", "attraction", "camp_site", "hotel")) {
         kind = sf.getString("tourism");
+        min_zoom = 15;
+      } else {
+        // Avoid problem of too many "other" kinds
+        // All these will default to min_zoom of 15
+        // If a more specific min_zoom is needed (or sanitize kind values)
+        // then add new logic in section above
+        if (sf.hasTag("amenity")) {
+          kind = sf.getString("amenity");
+        } else if (sf.hasTag("craft")) {
+          kind = sf.getString("craft");
+        } else if (sf.hasTag("aeroway")) {
+          kind = sf.getString("aeroway");
+        } else if (sf.hasTag("historic")) {
+          kind = sf.getString("historic");
+        } else if (sf.hasTag("landuse")) {
+          kind = sf.getString("landuse");
+        } else if (sf.hasTag("leisure")) {
+          kind = sf.getString("leisure");
+        } else if (sf.hasTag("railway")) {
+          kind = sf.getString("railway");
+        } else if (sf.hasTag("shop")) {
+          kind = sf.getString("shop");
+        } else if (sf.hasTag("tourism")) {
+          kind = sf.getString("tourism");
+        // Boundary is most generic, so place last else we loose out
+        // on nature_reserve detail versus all the protected_area
+        } else if (sf.hasTag("boundary")) {
+          kind = sf.getString("boundary");
+        }
+      }
+
+      if( sf.hasTag("boundary", "national_park") &&
```

Comment:
Special logic for `national_park`.

We'd also want handling for `national_forest`...